### PR TITLE
WIP - tests (issue #9)

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

**Comment by geoand** [comment]
@metacosm Can you please have a look at these changes?

I haven't actually integrated the execution of the tests into the main script, but I have altered the test script to allow it to be run either in standalone mode, or by being `sourced`-ed by another script, so we should be able to add it to the main script if we so choose.

File: run_tests.sh [hunk 1]
```diff
@@ -4,38 +4,78 @@
...
+        echo "Successfully tested ${canonical_name}"
+        #Delete the project since there is no need to inspect the results when everything is OK
+        oc delete project ${canonical_name}
     else
-        echo "Tests of ${booster} failed"
-        failed += ( ${booster} )
+        echo "Tests of ${canonical_name} failed"
+        failed_tests+=( ${canonical_name} )
+
+        #We don't delete the project because it could be needed for a postmortem inspection
```
[the remark this answers is not in the document]
Comment (geoand) [reply run_tests.sh]: My thinking was the user would do that by hand, once their postmortem inspection would be done

[the remark this answers is not in the document]
Comment (metacosm) [reply run_tests.sh]: OK.

File: run_tests.sh [hunk 2]
```diff
@@ -4,38 +4,78 @@
...
+    popd > /dev/null
+}
 
-echo "Done testing"
-if [ ${#failed[@]} -eq 0 ]; then
-    echo "All tests passes"
-else
-    echo "The following tests failed: "$(IFS=,; echo "${failed[*]}")
+# Execute the tests if this script was called directly (i.e. not sourced)
+if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
```
Comment (metacosm) [inline run_tests.sh]: Didn't know that trick!

File: run_tests.sh [hunk 3]
```diff
@@ -4,38 +4,78 @@
...
+
+        #We don't delete the project because it could be needed for a postmortem inspection
     fi
 
     cd ../
-done
+}
+
+execute_all_tests() {
+    pushd $(mktemp -d) > /dev/null
```
Comment (metacosm) [inline run_tests.sh]: Shouldn't we also remove the temp dir when we're done? See https://github.com/snowdrop/booster-common/blob/master/for-all-boosters.sh#L24-L40 as a way to do so.

File: run_tests.sh (hunk 2 above)
Comment (geoand) [reply run_tests.sh]: Very Python-esque :P

File: run_tests.sh (hunk 3 above)
Comment (geoand) [reply run_tests.sh]: Good point

**Pull request merged**